File: py/prime.py
def prime(n):
    if n%2 == 0:
        return False
    i = 3
    while i*i < n:
        if n%i == 0:
            return False
        i += 2
    return True

def sieve(N):
    s = [0,0,1]+[1,0]*(N/2)
    i = 3
    while i*i < N:
        if s[i]:
            for itr in xrange(i*2,N,i):
                s[itr] = 0
        i += 2
    return [i for i in range(N) if s[i]==1]

File: cpp/helper/prime.cpp
#include <vector>

bool prime(int n) {
    if (n == 2) {
        return true;
    } else if (n < 2 || n%2 == 0) {
        return false;
    } else {
        for (int i = 3; i*i <= n; i += 2) {
            if (n%i == 0) {
                return false;
            }
        }
        return true;
    }
}

std::vector<int> sieve(int n) {
    std::vector<int> s (n+1,1);
    s[0] = 0;
    s[1] = 0;
    int i, j;
    for (i = 4; i <= n; i += 2) {
        s[i] = 0;
    }
    for (i = 3; i*i <= n; i += 2) {
        if (s[i] == 1) {
            for (j = 2*i; j <= n; j += i) {
                s[j] = 0;
            }
        }
    }
    return s;
}
std::vector<int> primes_up_to(int n) {
    vector<int> s;
    s = sieve(n);
    std::vector<int> p;
    for (i = 0; i <= n; i++) {
        if (s[i] == 1) {
            p.push_back(i);
        }
    }
    return p;

}
